src/main/java/at/hannibal2/skyhanni/features/fishing/SeaCreatureFeatures.kt

```diff
@@ -81,14 +83,20 @@ object SeaCreatureFeatures {
 
     @HandleEvent(onlyOnSkyblock = true)
     fun onSeaCreatureFish(event: SeaCreatureFishEvent) {
-        if (!config.alertOwnCatches) return
+        if (!config.alertOwnCatches && !config.announceRare) return
 
         if (event.seaCreature.rare) {
-            val text = if (config.creatureName) "${event.seaCreature.displayName}!"
-            else "${event.seaCreature.rarity.chatColorCode}RARE CATCH!"
-            TitleManager.sendTitle(text, height = 2.8, fontSize = 7f)
-            if (config.playSound) SoundUtils.playBeepSound()
-            lastRareCatch = SimpleTimeMark.now()
+            if (config.alertOwnCatches) {
+                val text = if (config.creatureName) "${event.seaCreature.displayName}!"
+                else "${event.seaCreature.rarity.chatColorCode}RARE CATCH!"
+                TitleManager.sendTitle(text, height = 2.8, fontSize = 7f)
+                if (config.playSound) SoundUtils.playBeepSound()
+                lastRareCatch = SimpleTimeMark.now()
+            }
+            if (config.announceRare && PartyApi.partyMembers.size > 0) {
```

Comment:
instead of checking the party member size manually, please create a function in party api "isInParty" and use that instead here